supertokens_python/recipe/session/utils.py

```diff
@@ -323,7 +326,7 @@ def __init__(
         self,
         refresh_token_path: NormalisedURLPath,
         cookie_domain: Union[None, str],
-        cookie_same_site: Literal["lax", "strict", "none"],
+        cookie_same_site: Callable[[BaseRequest, Any], Awaitable[str]],
```

Comment:
Can you try using `Literal["lax", "strict", "none"]` instead of `str`? Maybe store the literal type in a variable and use it everywhere?